pkg/util/validation/limits.go

```diff
@@ -778,8 +782,32 @@ func (o *Overrides) IgnoreOOOExemplars(userID string) bool {
 	return o.getOverridesForUser(userID).IgnoreOOOExemplars
 }
 
+// ActiveSeriesCustomTrackersConfig returns all active series custom trackers that should be used for
+// the input tenant. The trackers are the merge of the configure base and additional custom trackers.
 func (o *Overrides) ActiveSeriesCustomTrackersConfig(userID string) asmodel.CustomTrackersConfig {
-	return o.getOverridesForUser(userID).ActiveSeriesCustomTrackersConfig
+	// Check if we already have a cached version of the merged ones.
+	if merged := o.getOverridesForUser(userID).activeSeriesMergedCustomTrackersConfig; merged != nil {
```

Comment:
If you store `o.getOverridesForUser(userID)` into a var, we can save a lot of code (we don't need the two getter below, for example)

Reply:
Addressed as part of https://github.com/grafana/mimir/pull/10428/commits/877b9ab0f2f47114933af7e42ddac52eaa608583